File: 0938-range-sum-of-bst/0938-range-sum-of-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
var rangeSumBST = function(root, low, high) {
    if(!root) return 0;

    const current = (root.val >= low && root.val <= high) ? root.val : 0;

    const left = rangeSumBST(root.left, low, high);
    const right = rangeSumBST(root.right, low, high);

    return current + left + right;
};

File: 0938-range-sum-of-bst/0938-range-sum-of-bst.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func rangeSumBST(root *TreeNode, low int, high int) int {
    if root == nil || root.Val == 0 {
        return 0
    }
    
    current := func(val int) int {
        if val <= high && val >= low {
            return val
        }
        return 0
    }(root.Val)
    
    left := rangeSumBST(root.Left, low, high)
    right := rangeSumBST(root.Right, low, high)

    return current + left + right
}